sqlglot/optimizer/scope.py

```diff
@@ -92,13 +96,17 @@ def clear_cache(self):
         self._pivots = None
         self._references = None
 
-    def branch(self, expression, scope_type, chain_sources=None, **kwargs):
+    def branch(
+        self, expression, scope_type, sources=None, cte_sources=None, lateral_sources=None, **kwargs
+    ):
         """Branch from the current scope to a new, inner scope"""
         return Scope(
             expression=expression.unnest(),
-            sources={**self.cte_sources, **(chain_sources or {})},
+            sources=sources.copy() if sources else {},
```

Comment:
just pass None here because you already instantiate {} on empty dict, same with lateral_sources